### PR TITLE
[rllib] example and docs on how to use parametric actions with DQN / PG algorithms

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Add examples of how to work with parametric action spaces (e.g., OpenAI 5 style).

## Related issue number

Closes https://github.com/ray-project/ray/issues/3364

cc @zegerhoogeboom

### CODE REVIEW

**Comment by zegerhoogeboom** [comment]
Running the cartpole example with `--run DQN` crashes with this error: 
```
ValueError: ('This model assumes num outputs is equal to max avail actions', 1, <tf.Tensor 'default/q_func/Reshape_1:0' shape=(?, 10, 2) dtype=float32>)
```

This is due to the `num_outputs` being hardcoded to be `1` in the DQN QNetwork

```python
def _build_q_network(self, obs, space):
        qnet = QNetwork(
            ModelCatalog.get_model({
                "obs": obs
            }, space, 1, self.config["model"]), self.num_actions,
```

In PPO, the logit dimension is determined by the action distribution which correctly returns `logit_dim = 10` in the example.
```python
dist_cls, logit_dim = ModelCatalog.get_action_dist(action_space, self.config["model"])
```
I don't know whether plugging this `logit_dim` into the QNetwork is conceptually correct though.

**Comment by ericl** [comment]
That should be fixed in the latest update -- try pulling.

**Comment by ericl** [comment]
Ping @richardliaw

**Review by richardliaw — APPROVED** [review]
`tf.boolean_mask(x, tf.logical_not(tf.is_inf(x)))` could be cleaner (for future ref)

File: python/ray/rllib/models/preprocessors.py [hunk 1]
```diff
@@ -164,6 +165,8 @@ def _init_shape(self, obs_space, options):
         return (size, )
 
     def transform(self, observation):
+        if not isinstance(observation, OrderedDict):
+            observation = OrderedDict(sorted(list(observation.items())))
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/models/preprocessors.py]: No, note that that is a check against the space spec, this is sorting the observation dict.

**Review by ericl — COMMENTED** [review]
The reason is_inf won't work is because inf is numerically unstable, so we use tf.float32.min instead

**Pull request merged**